test/token-service/test_tokenCreateTransaction.js

```diff
@@ -0,0 +1,8187 @@
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewPeriod: 0,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_RENEWAL_PERIOD");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#2) Creates a token with an auto renew period set to -1 seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewPeriod: -1,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_RENEWAL_PERIOD");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#3) Creates a token with an auto renew period of 9,223,372,036,854,775,807 (`int64` max) seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewPeriod: 9223372036854775807n,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_RENEWAL_PERIOD");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#4) Creates a token with an auto renew period of 9,223,372,036,854,775,806 (`int64` max - 1) seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewPeriod: 9223372036854775806n,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_RENEWAL_PERIOD");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#5) Creates a token with an auto renew period of 9,223,372,036,854,775,808 (`int64` max + 1) seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewPeriod: 9223372036854775808n,
```

Comment:
I am able to pass Duration of `9223372036854775808` seconds and it is passing in Java, how does it work in JS and c++ @ivaylogarnev-limechain 
 @rwalworth ?

Reply:
I haven't tested it in JS but in C++ its being handled by the server.

Reply:
hmmm, that value should be rejected by consensus as invalid; it parses to a negative number in the Java code.
The question is what is translating that Duration to an integer for the API, and is that SDK code applying a boundary limit.

Reply:
Oh yes, it's actually overflowing.
The question now is now to pass a number that exceeds the int64 boundary as a int64 parameter :D

Reply:
So we want it to overflow and the consensus would return an error because the value is negative?
Maybe some of the java util methods are doing some magic under the hood to cause weird behaviour, will check it.

Reply:
Yes.
Protobuf handles unsigned gracefully for Java (and we want the API to use unsigned for values that are never negative), but Java itself has no unsigned support, so the overflow case generally _should_ be tested and verified.